RETAIL-PLAN.md

```diff
@@ -0,0 +1,58 @@
+
+### Part 1: Crowdsourcing Voice Models (ALREADY IN PROGRESS)
+
+Develop a user-friendly website to facilitate the crowdsourcing of voice models for all races in World of Warcraft. The website should include the following features:
+
+- User registration and authentication to enable contributors to sign up and submit voice samples
+- Clear instructions and guidelines on how to provide high-quality voice samples, including preferred recording setup, script examples, and ideal pronunciation
+- A submission system for users to upload their voice samples, including categorization by race, gender, and any other relevant attributes
+- An upvoting or rating system for users to evaluate the quality of submitted voice samples and provide feedback to contributors
+- A progress tracker to monitor the development of voice models for each race and gender, encouraging users to contribute to underrepresented categories
+
+### Part 2: Collecting Quest/Gossip Text Data (LOOKING FOR HELP)
+
+#### New Feature in WoW Voiceover Addon
+
+- Add a new feature to the WoW Voiceover addon that enables users to opt-in to collect quest/gossip text data during gameplay
+- Create a lightweight companion executable that runs in the background during gameplay to upload the scraped text data to a central server
+- The data should include:
+
+| Column         | Description                                                |
+|----------------|------------------------------------------------------------|
+| source         | Type of interaction ('accept', 'progress', 'complete', or 'gossip') |
+| quest          | Quest ID or empty string if it's a gossip interaction      |
+| text           | text content of the interaction |
+| race           | Race of the NPC involved in the interaction                 |
+| sex            | Gender of the NPC involved in the interaction               |
+| npc_name           | Name of the NPC involved in the interaction                 |
+| npc_type           | Type of the NPC involved ('creature', 'gameobject', or 'item') |
+| npc_id             | Creature/gameobject/item ID of the NPC involved in the interaction |
+| player_race    | Race of the player involved in the interaction              |
+| player_gender  | Gender of the player involved in the interaction            |
+| player_name  | Name of the player involved in the interaction            |
+
+- Race and Sex will probably not be possible to collect directly from the game addon for recent/current expansions.
+
+#### API and Database Development
+
+- Develop a robust API and database to store and manage the collected text data
+- The API should provide endpoints for data retrieval, updates, and submission of new data from the companion executable
+- Seed the database with initial data from WoWhead and other sources, and continually update it with scraped data from users' gameplay
```

Comment:
It would be useful to also keep track of which expansion does the quest/creature belong to. This should be possible to obtain for quests using scraped wowhead data, as they seem to be keeping track of it and are displaying it on appropriate pages ("Added in patch x.y.z").

For gossip this wouldn't be possible, tho. While we could also use wowhead's data to assign creatures to expansions, a creature from an older expansion can have a new gossip text/options added in a later expansion.

Reply:
done